### PR TITLE
Test

### PR DESCRIPTION
**Pull request by s4826**
- Remove duplicated code in ```fixed_horizon_delta()```
- updated tests to reflect new method signature for ```fixed_horizon_delta()```

### CODE REVIEW

**Review by jbao — APPROVED** [review]

**Comment by mkolarek** [comment]
:+1:

**Review by mkolarek — APPROVED** [review]

File: expan/core/experiment.py [hunk 1]
```diff
@@ -399,9 +400,14 @@ def delta(self, method='fixed_horizon', kpi_subset=None, derived_kpis=None, **kw
...
 			f, a1, a2 = m[0], m[1], m[2]
-			return f(a1, a2, **kwargs)
+			if method == 'fixed_horizon':
+				# pass initialized results object into fixed_horizon_delta
+				return f(res, a1, a2, **kwargs)
+			else:
+				return f(a1, a2, **kwargs)
 
-	def fixed_horizon_delta(self, 
+	def fixed_horizon_delta(self,
```
Comment (gbordyugov) [inline expan/core/experiment.py:409]: Is introducing an additional parameter to fixed_horizon_delta() worth it? It seems to me that this introduces an inhomogeneity across the fourth `delta` workers

File: tests/tests_core/test_results.py [hunk 2]
```diff
@@ -37,6 +38,34 @@ def load_example_results():
 
 	return r.from_hdf(example_fpath)
 
+def mock_results_object(data, kpi_subset=None, derived_kpis=None, **kwargs):
+	"""
+	Create a results object for any tests involving fixed_horizon_delta()
```
Comment (gbordyugov) [inline tests/tests_core/test_results.py:43]: why factor out this functionality as a separate function? is it going to be used anywhere besides `fixed_horizon_delta()`?

addendum: would this code persist in Experiment.delta() further?

**Review by gbordyugov — CHANGES_REQUESTED** [review]
Hi,

great job! I'm however still a bit unconvinced about your solution to move the duplicate code into the `mock` function. Does it eliminate the duplicate code in both `delta()` and `fixed_horizon_delta()`?

Plus introducing a `res` argument to `fixed_horizon_delta()` makes its handling a bit special in comparison to other workers.

Best,
gb

**Review by gbordyugov — APPROVED** [review]
Hi,

I totally misunderstood the logic of what you did, now, after a bit of pondering, it looks great to me.

**Comment by jbao** [comment]
This resolves #81 .

**Pull request merged**